docs/extended_api/memory_model.md

```diff
@@ -0,0 +1,186 @@
+---
+parent: Extended API
+nav_order: 0
+---
+
+# Memory model
+
+Standard C++ presents a view that the cost to synchronize threads is uniform and low.
+
+CUDA C++ is different: the cost to synchronize threads grows as threads are further appart.
+It is low across threads within a block, but high across arbitrary threads in the system running on multiple GPUs and CPUs.
+
+To account for non-uniform thread synchronization costs that are not always low, CUDA C++ extends the standard C++ memory model and concurrency facilities in the `cuda::` namespace with **thread scopes**, retaining the syntax and semantics of standard C++ by default.
+
+## Threads
+
+In CUDA C++, memory can only be modified by threads, so any operation that modifies memory is being executed by a thread.
+
+Examples of threads of execution in CUDA C++ programs: GPU or CPU threads, threads in DMA engines executing asynchronous operations like [`memcpy_async`] or [`cudaMemcpyAsync`], threads in NICs executing RDMA operations, etc.
```

Comment:
NACK this new Threads subsection.  We shouldn't talk about DMA engines, NICs, etc. here, and we shouldn't claim that nothing outside a thread can modify memory.

Reply:
This section does not claim "that nothing outside a thread can modify memory." but rather that those operations modify memory via a C++ thread.

I can modify this sentence to:

```suggestion
Examples of threads of execution in CUDA C++ programs: GPU threads, CPU threads, thread asynchronously copying memory for [`memcpy_async`] or [`cudaMemcpyAsync`], etc.
```

Reply:
Leaving sentence out: 

```suggestion
```